lib/src/main/java/com/auth0/android/lock/views/AuthConfig.java

```diff
@@ -0,0 +1,196 @@
+package com.auth0.android.lock.views;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.ColorInt;
+import android.support.annotation.NonNull;
+import android.support.annotation.StyleRes;
+import android.support.annotation.StyleableRes;
+import android.support.v4.content.ContextCompat;
+
+import com.auth0.android.lock.R;
+
+public class AuthConfig {
+
+    private String strategyName;
+    private String connectionName;
+    @StyleRes
+    private int styleRes;
+
+    public AuthConfig(@NonNull String strategyName, @NonNull String connectionName, @StyleRes int styleRes) {
+        this.strategyName = strategyName;
+        this.connectionName = connectionName;
+        this.styleRes = styleRes;
+    }
+
+    @NonNull
+    public String getConnectionName() {
+        return connectionName;
+    }
+
+    @NonNull
+    public String getName(@NonNull Context context) {
+        //noinspection ResourceType
+        int id = getIdForResource(context, R.attr.Auth0_Name);
+        return id != -1 ? context.getString(id) : strategyName;
+    }
+
+    @NonNull
+    public Drawable getLogo(@NonNull Context context) {
+        //noinspection ResourceType
+        int id = getIdForResource(context, R.attr.Auth0_Logo);
+        if (id == -1) {
+            id = R.drawable.com_auth0_lock_ic_social_auth0;
+        }
+        return ContextCompat.getDrawable(context, id);
+    }
+
+    @ColorInt
+    public int getBackgroundColor(@NonNull Context context) {
+        //noinspection ResourceType
+        int id = getIdForResource(context, R.attr.Auth0_BackgroundColor);
+        if (id == -1) {
+            id = R.color.com_auth0_lock_social_unknown;
+        }
+        return ContextCompat.getColor(context, id);
+    }
+
+    /**
+     * Retrieves the resource id of the given Style index.
+     *
+     * @param context a valid Context
+     * @param index   The index to search on the Style definition.
+     * @return the id if found or -1.
+     */
+    int getIdForResource(@NonNull Context context, @StyleableRes int index) {
+        final int[] attrs = new int[]{index};
```

Comment:
Maybe it's better to use 
```TypedArray obtainStyledAttributes (AttributeSet set, int[] attrs, int defStyleAttr, int defStyleRes)```
at the constructor, using the default Theme as param, like its done in the custom views. 
Would be less code and also the resources would only be looked up once. We'll need to add a context to the constructor.